src/application/components/ClientNotFoundModal.tsx

```diff
@@ -0,0 +1,208 @@
+            </Disclosure.Panel>
+          </Disclosure>
+
+          <Disclosure>
+            <Disclosure.Button>
+              I have set <code>connectToDevTools</code> to <code>true</code>
+            </Disclosure.Button>
+            <Disclosure.Panel>
+              <p>
+                Apollo Client Devtools relies on the presence of a global{" "}
+                <code>window.__APOLLO_CLIENT__</code> variable set from the
+                Apollo Client instance. Check your browser console for{" "}
+                <code>window.__APOLLO_CLIENT__</code> to see if this variable is
+                set.
+              </p>
+              <p className="mt-4">
+                If the value is <code>undefined</code>, this is most likely a
+                bug with Apollo Client. Please open a{" "}
+                <GitHubIssueLink
+                  labels={["🐞 bug"]}
+                  repository="apollo-client"
+                  body={`
+<!-- Please provide a detailed description of the issue you are experiencing. It is most helpful if you are able to provide a minimal reproduction of the issue. -->
+
+### Link to Reproduction
+<!-- Please provide a link to the reproduction of the issue. -->
+
+${SECTIONS.apolloClientVersion}
+${SECTIONS.devtoolsVersion}
+`}
+                >
+                  GitHub issue
+                </GitHubIssueLink>{" "}
+                in the Apollo Client repository to get help from the Apollo
+                Client maintainers.
+              </p>
+            </Disclosure.Panel>
+          </Disclosure>
+
+          <Disclosure>
+            <Disclosure.Button>
+              <code>window.__APOLLO_CLIENT__</code> references my client
+              instance
+            </Disclosure.Button>
+            <Disclosure.Panel>
+              <p>
+                Occasionally the Apollo Client instance was set correctly on
+                <code>window</code>, but the Apollo Client Devtools could not
+                find it in time. Apollo Client Devtools checks for the presence
+                of the <code>window.__APOLLO_CLIENT__</code> variable for up to
+                10 seconds before giving up.
+              </p>
+              <p className="mt-4">
+                Apollo Client Devtools exposes a global{" "}
+                <code>window.__APOLLO_DEVTOOLS_GLOBAL_HOOK__</code> variable
+                that you can use to check if the client was discovered. Check
+                your browser console for the{" "}
+                <code>window.__APOLLO_DEVTOOLS_GLOBAL_HOOK__.ApolloClient</code>{" "}
+                variable to see if it references your Apollo Client instance. If
+                this value is <code>undefined</code>, this likely indicates your
+                client was not discovered in time. Click the &quot;Retry&quot;
+                button below to try connecting to your Apollo Client instance
+                again.
+              </p>
+              <p className="mt-4">
```

Comment:
The only thing I'd add here is this also occurs when using our Next.js experimental package if the first page(s) you load don't render any client components that initialize AC in the first 10 seconds: https://github.com/apollographql/apollo-client-devtools/issues/1035

We may want to mention that case specifically since you can now just click reload after navigating, and close that issue 🎉

Reply:
Thats a good callout! I think one more "I'm using Next.js" disclosure is probably a good idea 🙂

Reply:
Done in [8859408](https://github.com/apollographql/apollo-client-devtools/pull/1220/commits/8859408649cdcc678651630c45cfcd187e1f8b56). Please let me know if this captures the spirit of what is happening here 🙂